python/revng/daemon/schema.py

```diff
@@ -214,6 +214,18 @@ async def resolve_step_analyses(step_obj, info):
     return await run_in_executor(lambda: [a.as_dict() for a in step.analyses()])
 
 
+@step.field("artifacts")
+async def resolve_step_artifacts(step_obj, info):
+    manager: Manager = info.context["manager"]
+    step = await run_in_executor(lambda: manager.get_step(step_obj["name"]))
+    artifacts_container = await run_in_executor(step.get_artifacts_container)
+    artifacts_kind = await run_in_executor(step.get_artifacts_kind)
```

Comment:
1. in simple terms: a thread + a queue where you submit functions to run
2. Added a comment in the commit to describe the situation
3. If this was JS/TS I would, but python's `lambda` is really lackluster so it doesn't allow the flexibility of wrapping 3 calls in one (without defining a function within)
3b. also it would look a bit ugly with argument unpacking
```python
def task():
  step = manager.get_step(step_obj["name"]
  artifacts_container = step.get_artifacts_container()
  artifacts_kind = step.get_artifacts_kind()
  artifacts_single_target_filename = step.get_artifacts_single_target_filename()
  return step, artifacts_container, artifacts_kind, artifacts_single_target_filename
  
step, artifacts_container, artifacts_kind, artifacts_single_target_filename = await run_in_executor(task)
```